src/app/shared/data-classes-list/data-classes-list.component.ts

```diff
@@ -304,16 +317,7 @@ export class DataClassesListComponent implements AfterViewInit {
   }
 
   bulkDelete() {
-    const dataElementIdLst = [];
-    this.dataClassRecords.forEach((record) => {
-      if (record.checked) {
-        dataElementIdLst.push({
-          id: record.id,
-          domainType: record.domainType
-        });
-      }
-    });
-
+    const dataElementIdLst = this.dataClassRecords.filter(record => record.checked === true);
```

Comment:
Nice consolidation of code here. For the `filter()` predicate, you may even reduce it further to `filter(record => record.checked)`